spring-kafka/src/main/java/org/springframework/kafka/support/micrometer/KafkaTemplateObservation.java

```diff
@@ -90,13 +138,17 @@ public static class DefaultKafkaTemplateObservationConvention implements KafkaTe
 
 		@Override
 		public KeyValues getLowCardinalityKeyValues(KafkaRecordSenderContext context) {
-			return KeyValues.of(KafkaTemplateObservation.TemplateLowCardinalityTags.BEAN_NAME.asString(),
-							context.getBeanName());
+			return KeyValues.of(
+					TemplateLowCardinalityTags.BEAN_NAME.withValue(context.getBeanName()),
+					TemplateLowCardinalityTags.MESSAGING_SYSTEM.withValue("kafka"),
+					TemplateLowCardinalityTags.MESSAGING_OPERATION.withValue("publish"),
+					TemplateLowCardinalityTags.MESSAGING_DESTINATION_KIND.withValue("topic"),
+					TemplateLowCardinalityTags.MESSAGING_DESTINATION_NAME.withValue(context.getDestination()));
 		}
 
 		@Override
 		public String getContextualName(KafkaRecordSenderContext context) {
-			return context.getDestination() + " send";
+			return context.getDestination() + " publish";
```

Comment:
I think, Gary, when they asked us for this, the OTEL really promoted a `send`.
The `publish` has come later on.

I might be mistaken. Let's double check with @marcingrzejszczak!

Reply:
Even if we change it; it has to be an option; I don't think we can make a change like this in a patch release.

Reply:
Presumably, we are going to need something similar for RabbitMQ too.

Reply:
Well, I think that `publish`/`send` is still just a recommendation.

I suggest to stick with a `send` so far until the next `3.1` this Fall.
If other change are additive and don't override anything existing, we can apply them with this PR.

Reply:
We never said that we are going to be OpenTelemetry compliant. OTel has a tendency to do breaking changes in semantic conventions in minor / patch releases. This is not the way that we would like to go with. At the time of doing this instrumentation AFAIR the OTel recommendation was `send` and I think we should keep it that way. One can use a `ObservationFilter` to change the name from `send` to `publish` if they want to be compliant with the current version of OTel semantic conventions.

Reply:
Thank you, Marcin!

So, our conclusion is like this then: we accept this change, but not with a `publish` move. We stick with `send` for time being.

I think such a statement about an OTel compliant must be mentioned somewhere in the Micrometer Tracing docs: https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup.

Reply:
> So, our conclusion is like this then: we accept this change, but not with a publish move. We stick with send for time being.

Correct!

> I think such a statement about an OTel compliant must be mentioned somewhere in the Micrometer Tracing docs: https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup.

We were looking at the OTel semantic conventions at the point of time where things got instrumented but we were never saying that we want to be OTel compliant because of the reasons I mentioned above. We never explicitly stated that we are OTel compliant really so I don't think we should tell people about it.

Also naming is unrelated to the tracer library used under the hood. So naming can be compliant with OTel semantic conventions even if you're using Brave.

Reply:
Sure! No problem!
But why then people push for Otel everywhere? 😄 
It feels like they took an idea for all of these tags to be in line with Otel from somewhere on top.

See the heading of this PR, for example:

> Opentelemetry defines standards for specifying and providing proper information for tracing spans

If we don't say anywhere in our docs that we follow so-called standards then how people came to such a conclusion?